### PR TITLE
Migrated gradle remote cache to access key authentication

### PR DESCRIPTION
**Pull request by ribafish**
Migrated gradle remote cache to the gradleEnterprise connector and with that to access key authentication.

### CODE REVIEW

File: src/main/java/io/micronaut/build/MicronautGradleEnterprisePlugin.java [hunk 1]
```diff
@@ -82,20 +80,14 @@ private void configureGradleEnterprise(Settings settings,
...
+                            String accessKey = providers.systemProperty("GRADLE_ENTERPRISE_ACCESS_KEY").getOrNull();
+                            if (accessKey != null && !accessKey.isEmpty()) {
                                 remote.setPush(true);
-                                remote.credentials(creds -> {
-                                    creds.setUsername(username);
-                                    creds.setPassword(password);
-                                });
                             } else {
-                                System.err.println("WARNING: No credentials for remote build cache, cannot configure push!");
+                                System.err.println("WARNING: Access key missing for remote build cache, cannot configure push!");
```
Comment (runningcode) [inline src/main/java/io/micronaut/build/MicronautGradleEnterprisePlugin.java:90]: I think the string can be left the same. They are both still credentials at the end of the day.

File: src/main/java/io/micronaut/build/MicronautGradleEnterprisePlugin.java [hunk 2]
```diff
@@ -82,20 +80,14 @@ private void configureGradleEnterprise(Settings settings,
...
-                        remote.setUrl("https://ge.micronaut.io/cache/");
+                    buildCache.remote(ge.getBuildCache(), remote -> {
                         remote.setEnabled(true);
                         if (push) {
-                            String username = envOrSystemProperty(providers, "GRADLE_ENTERPRISE_CACHE_USERNAME", "gradleEnterpriseCacheUsername", "");
-                            String password = envOrSystemProperty(providers, "GRADLE_ENTERPRISE_CACHE_PASSWORD", "gradleEnterpriseCachePassword", "");
-                            if (!username.isEmpty() && !password.isEmpty()) {
+                            String accessKey = providers.systemProperty("GRADLE_ENTERPRISE_ACCESS_KEY").getOrNull();
+                            if (accessKey != null && !accessKey.isEmpty()) {
                                 remote.setPush(true);
```
Comment (runningcode) [inline src/main/java/io/micronaut/build/MicronautGradleEnterprisePlugin.java:88]: What about this?
```suggestion
                                remote.setPush(accessKey != null && !accessKey.isEmpty());
```

Comment (ribafish) [reply src/main/java/io/micronaut/build/MicronautGradleEnterprisePlugin.java:88]: Yes, I thought about doing that, however if we leave the `if` in place...I don't see a reason for that, and without the `if`, we can't report it to the user.

File: src/main/java/io/micronaut/build/MicronautGradleEnterprisePlugin.java (hunk 1 above)
Comment (ribafish) [reply src/main/java/io/micronaut/build/MicronautGradleEnterprisePlugin.java:90]: That is correct, however changing it is clearer for the user as to what's missing.

File: src/main/java/io/micronaut/build/MicronautGradleEnterprisePlugin.java (hunk 2 above)
Comment (runningcode) [reply src/main/java/io/micronaut/build/MicronautGradleEnterprisePlugin.java:88]: Ah i see what you mean. Looks good then!

**Review by runningcode — APPROVED** [review]

**Comment by ribafish** [comment]
Superseded by https://github.com/micronaut-projects/micronaut-build/pull/653

**Pull request closed without merging**